File: Problems 0100-0199/128. Longest Consecutive Sequence/solution.cpp
class Solution {
    public:
        int longestConsecutive(vector<int>& nums) {
            unordered_set<int> unique(nums.begin(), nums.end());
            int res = 0;
            for (int num : unique) {
                // if num is smallest in its consecutive path
                int cand = num;
                if (unique.find(cand-1) == unique.end()) {
                    int count = 1;
                    while (unique.find(cand+1) != unique.end()) {
                        cand++;
                        count++;
                    }
                    res = max(res, count);
                }
            }
    
            return res;
    
        }
    };

File: Problems 0100-0199/128. Longest Consecutive Sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        unique = set(nums)

        res = 0
        for num in unique:
            if num-1 not in unique:
                conseq = 1
                while num+1 in unique:
                    num += 1
                    conseq += 1
                res = max(res, conseq)
        return res